testing/trino-testing/src/main/java/io/trino/testing/QueryAssertions.java

```diff
@@ -336,7 +336,7 @@ private static void assertDistributedQuery(
         List<MaterializedRow> actualRows = actualResults.getMaterializedRows();
         List<MaterializedRow> expectedRows = expectedResults.getMaterializedRows();
 
-        if (compareUpdate) {
+        if (compareUpdate && !actualResults.getUpdateType().equals(Optional.of("ALTER TABLE EXECUTE"))) {
```

Comment:
Instead of skipping here, can we introduce something `assertAlter` or a dedicated method - which would set `compareUpdate` to false ?

Reply:
There should be no need for a special `assertAlter` method. We should be able to assert a more detailed output with assertUpdate itself.
This branch is written with the assumption that an update query can only produce a row count output and nothing else. Since ALTER commands did not output anything before, this branch was never executed before.
Now with the more detailed output, we need to skip over this branch to allow the assertion on detailed results.